app/components/LoginPage.tsx

```diff
@@ -1,17 +1,62 @@
-// @ts-nocheck File not migrated fully to TS
-
-import _ from 'lodash';
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
+import { get, isEmpty } from 'lodash';
 import { parse } from 'query-string';
-import i18n from 'i18next';
-import { Button, Input, Message, Form, FullScreenSegment, Logo } from './basic';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import styled from 'styled-components';
+
+import type { Dispatch } from 'redux';
+import type { ReduxState } from '../reducers';
+
+import { login } from '../actions/managers';
 import SplashLoadingScreen from '../utils/SplashLoadingScreen';
-import LogoLabel from './banner/LogoLabel';
+import StageUtils from '../utils/stageUtils';
 import LargeLogo from './banner/LargeLogo';
+import LogoLabel from './banner/LogoLabel';
+import { Button, Form, FullScreenSegment, Input, Message } from './basic';
+
+export interface LoginPageProps {
+    isLoggingIn: boolean;
+    isSamlEnabled: boolean;
+    onLogin: (username: string, password: string, redirect?: string) => void;
+    location: {
+        search: string;
+    };
+    loginError: string | null;
+    samlSsoUrl: string;
+    username: string;
+    whiteLabel: {
+        loginPageHeaderColor: string;
+        loginPageTextColor: string;
+    };
+}
 
-export default class LoginPage extends Component {
-    constructor(props, context) {
+interface LoginPageState {
+    username: string;
+    password: string;
+    errors: Errors;
+}
+
+type Errors = {
+    username?: string;
+    password?: string;
+} | null;
+
+const StyledInput = styled(Input)`
+    &&&&&& input:autofill,
+    input:autofill:focus,
+    input:autofill:hover,
+    input:autofill:focus,
```

Comment:
After refactoring, some code duplication has sneak into the selectors 😁 
![additionalcode](https://user-images.githubusercontent.com/89005877/178472587-6ecca68a-c51f-4ae5-a867-cba6d74d8712.png)

```suggestion
    input:autofill:focus,
    input:autofill:hover,
```